common/src/main/java/me/shedaniel/architectury/event/events/PlayerEvent.java

```diff
@@ -70,62 +105,179 @@
         void pickup(Player player, ItemEntity entity, ItemStack stack);
     }
     
     interface ChangeDimension {
+        /**
+         * Invoked when a player changes their dimension.
+         * Equivalent to Forge's {@code PlayerChangedDimensionEvent} event.
+         *
+         * @param player   The teleporting player.
+         * @param oldLevel The level the player comes from.
+         * @param newLevel The level the player teleports into.
+         */
         void change(ServerPlayer player, ResourceKey<Level> oldLevel, ResourceKey<Level> newLevel);
     }
     
     interface DropItem {
+        /**
+         * Invoked when a player drops an item.
+         * Equivalent to Forge's {@code ItemTossEvent} event.
+         *
+         * @param player The player dropping something.
+         * @param entity The entity that has spawned when the player dropped a ItemStack.
+         * @return Forge ignores the return value. On Fabric a {@link InteractionResult#FAIL} results in the event getting canceled and the item not being dropped.
+         */
         InteractionResult drop(Player player, ItemEntity entity);
     }
     
     interface BreakBlock {
+        /**
+         * Deprecated variant of {@link BlockEvent#BREAK}. Defers to the proper event.
+         */
         InteractionResult breakBlock(Level world, BlockPos pos, BlockState state, ServerPlayer player, @Nullable IntValue xp);
     }
     
     interface OpenMenu {
+        /**
+         * Invoked when a player opens a menu.
+         * Equivalent to Forge's {@code PlayerContainerEvent.Open} event.
+         *
+         * <p>If you want to prevent a certain menu to be opened, you have to catch the {@link me.shedaniel.architectury.event.events.GuiEvent#SET_SCREEN} event and open your own.
+         *
+         * @param player The player opening the menu.
+         * @param menu   The menu that is opened.
+         */
         void open(Player player, AbstractContainerMenu menu);
     }
     
     interface CloseMenu {
+        /**
+         * Invoked when a player closes a menu.
+         * Equivalent to Forge's {@code PlayerContainerEvent.Close} event.
+         *
+         * @param player The player closing the menu.
+         * @param menu   The menu that is closed.
+         */
         void close(Player player, AbstractContainerMenu menu);
     }
     
     interface FillBucket {
+        /**
+         * Invoked when a player attempts to fill a bucket using right-click.
+         * You can return a non-PASS interaction result to cancel further processing by other mods.
+         *
+         * @param player The player filling the bucket.
+         * @param level  The level the player is in.
+         * @param stack  The bucket stack.
+         * @param target The target which the player has aimed at.
+         * @return On Forge, FAIL cancels the event, and SUCCESS sets the event as handled. On Fabric, any non-PASS result is returned directly and immediately.
```

Comment:
No such thing of FAIL, SUCCESS and PASS for CompoundEventResult, please just shorten this to the general meaning of the parameter.